network/dag/state.go

```diff
@@ -134,6 +149,8 @@ func (s *state) Add(ctx context.Context, transaction Transaction, payload []byte
 		if emitPayloadEvent {
 			s.notify(payloadEvent)
 		}
+	}), stoabs.AfterCommit(func() {
+		s.transactionCount.Inc()
```

Comment:
duplicate checking is done within the context of this transaction (L114). If the transaction already exists it exits, but still increments the counter

Reply:
good catch